CodeEditModules/Modules/Git/src/Client/Models/ChangedFiles.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  ChangedFiles.swift
+//  
+//
+//  Created by Nanashi Li on 2022/05/05.
+//
+
+import Foundation
+import SwiftUI
+import WorkspaceClient
+
+public struct ChangedFiles: Codable, Hashable, Identifiable {
+    public var id = UUID()
+    public var changeType: String
+    public var fileLink: String
+
+    /// Use it like this
+    /// ```swift
+    /// Image(systemName: item.systemImage)
+    /// ```
+    public var systemImage: String {
+        if !isFile {
+            return "folder.fill"
+        } else {
+            return FileIcon.fileIcon(fileType: fileType)
+        }
+    }
+
+    /// Returns the file name (e.g.: `Package.swift`)
+    public var fileName: String {
+        URL(string: fileLink)?.lastPathComponent ?? ""
+    }
+
+    private var isFile: Bool {
+        return fileName.contains(".")
```

Comment:
I'm not surely understanding the logic of this. If we have an URL we can have more solid logic to understand if it's a file or a directory

Reply:
yeah, but we should change the link to an URL and we should the URL API to check if it's a file or not